wanda/forge.go

```diff
@@ -161,26 +181,60 @@ func (f *Forge) Build(spec *Spec) error {
 	}
 	log.Println("build input digest: ", inputDigest)
 
+	cacheTag := f.cacheTag(inputDigest)
+	workTag := f.workTag(spec.Name)
+
 	// TODO(aslonnie): check if the image output already exists
 	// if yes, then just perform retag, rather than rebuilding.
+	if f.config.RayCI && f.config.WorkRepo != "" {
+		ct, err := cranename.NewTag(cacheTag)
+		if err != nil {
+			return fmt.Errorf("parse cache tag %q: %w", cacheTag, err)
+		}
+
+		wt, err := cranename.NewTag(workTag)
+		if err != nil {
+			return fmt.Errorf("parse work tag %q: %w", workTag, err)
+		}
+
+		desc, err := remote.Get(ct, f.remoteOpts...)
+		if err != nil {
+			log.Printf("fetch cache image: %v", err)
```

Comment:
the log should say 'failed to fetch cache image' or 'no cache image found'?

Reply:
changed to "Cache image miss"